optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/common/nearby/NearbyRandomFactory.java

```
@@ -0,0 +1,107 @@
+
+public class NearbyRandomFactory {
+
+    public static NearbyRandomFactory create(NearbySelectionConfig nearbySelectionConfig) {
+        return new NearbyRandomFactory(nearbySelectionConfig);
+    }
+
+    private final NearbySelectionConfig nearbySelectionConfig;
+
+    public NearbyRandomFactory(NearbySelectionConfig nearbySelectionConfig) {
+        this.nearbySelectionConfig = nearbySelectionConfig;
+    }
+
+    public NearbyRandom buildNearbyRandom(boolean randomSelection) {
+        boolean blockDistributionEnabled =
+                nearbySelectionConfig.getNearbySelectionDistributionType() == NearbySelectionDistributionType.BLOCK_DISTRIBUTION
+                        || nearbySelectionConfig.getBlockDistributionSizeMinimum() != null
+                        || nearbySelectionConfig.getBlockDistributionSizeMaximum() != null
+                        || nearbySelectionConfig.getBlockDistributionSizeRatio() != null
+                        || nearbySelectionConfig.getBlockDistributionUniformDistributionProbability() != null;
+        boolean linearDistributionEnabled = nearbySelectionConfig
+                .getNearbySelectionDistributionType() == NearbySelectionDistributionType.LINEAR_DISTRIBUTION
+                || nearbySelectionConfig.getLinearDistributionSizeMaximum() != null;
+        boolean parabolicDistributionEnabled = nearbySelectionConfig
+                .getNearbySelectionDistributionType() == NearbySelectionDistributionType.PARABOLIC_DISTRIBUTION
+                || nearbySelectionConfig.getParabolicDistributionSizeMaximum() != null;
+        boolean betaDistributionEnabled =
+                nearbySelectionConfig.getNearbySelectionDistributionType() == NearbySelectionDistributionType.BETA_DISTRIBUTION
+                        || nearbySelectionConfig.getBetaDistributionAlpha() != null
+                        || nearbySelectionConfig.getBetaDistributionBeta() != null;
```

How about:

    boolean distributionEnabled = false;
    switch (nearbySelectionConfig.getNearbySelectionDistributionType()) {
         case LINEAR_DISTRIBUTION:
              distributionEnabled = nearbySelectionConfig.getLinearDistributionSizeMaximum() != null;
              break;
         ...
    }

Less repetition, and in my opinion more clarity.

---

Let me just answer this and the next two comments together, as they are connected with each other:
The method consists of 3 steps:

1. discovers which distribution types are enabled
2. fails fast if multiple distribution types have been configured
3. returns the computed distribution

From your comments, I understand you suggest implementing all these steps in a single switch statement.

To implement step 2, the subsequent branches of the switch statement would have to depend on preceding ones. Each case would contain:
- the logic for discovering its distribution type
- fail fast condition(s)
- instantiation of the NearbyRandom

Since the implementation of fail fast conditions requires avoiding `break` after any `case`, we cannot use the default branch, which is supposed to return `LinearDistributionNearbyRandom(Integer.MAX_VALUE)`.

To sum this up, I don't think doing all 3 steps in the switch statement (while there will be additional if conditions after it anyway) is as an improvement. But I like the idea of using switch for the step 1.

---

Actually even using switch just for step 1 wouldn't help, as the individual conditions are connected with logical OR.